core/slice.go

```diff
@@ -217,22 +217,26 @@ func (sl *Slice) Append(header *types.Header, domPendingHeader *types.Header, do
 	// list of confirmed ETXs using the subordinate manifest In either case, if
 	// we are a dominant node, we need to collect the ETX rollup from our sub.
 	if !domOrigin && nodeCtx != common.ZONE_CTX {
-		newInboundEtxs, _, err = sl.CollectNewlyConfirmedEtxs(block, block.Location())
-		if err != nil {
-			log.Trace("Error collecting newly confirmed etxs: ", "err", err)
-			// Keeping track of the number of times pending etx fails and if it crossed the retry threshold
-			// ask the sub for the pending etx/rollup data
-			val, exist := sl.pEtxRetryCache.Get(block.Hash())
-			var retry uint64
-			if exist {
-				pEtxCurrent, ok := val.(pEtxRetry)
-				if ok {
-					retry = pEtxCurrent.retries + 1
+		newInboundEtxs = rawdb.ReadInboundEtxs(sl.sliceDb, block.Hash())
```

Comment:
why write to database instead of an in-memory cache? This data is not needed in depth (don't need database), and caches are faster.

Reply:
Yes, for Prime and Region cache would be more optimal, so will create a newInboundEtx cache